src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY_DEFAULT = 10;
+    private static final double CAPACITY_GROW = 1.5;
+    private int size;
+    private T[] arrayList;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[CAPACITY_DEFAULT];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == arrayList.length) {
+            grow();
+        }
+        arrayList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        if (size == arrayList.length) {
+            grow();
+        }
+        System.arraycopy(arrayList, index,arrayList, index + 1, size - index);
+        arrayList[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list == null) {
+            return;
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return arrayList[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        arrayList[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        final T removedArrayList = arrayList[index];
+        System.arraycopy(arrayList, index + 1, arrayList, index, size - 1 - index);
+        size--;
+        return removedArrayList;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < this.size; i++) {
+            if (arrayList[i] == element || arrayList[i] != null && arrayList[i].equals(element)) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException(" The list is empty");
```

Comment:
That does not mean our list is empty, only that we don't have such an element
```suggestion
        throw new NoSuchElementException("Element: " + element + "is not present in the list");
```